### PR TITLE
[gui] refactors modality handling for dialogs

### PR DESCRIPTION
**Pull request by xhaggi**
This PR refactors the modality handling of dialogs out of the methods `CGUIDialog::DoModal` and `CGUIDialog::Show`. Now we specify the modality on a per dialog basis and the modality is not changed depending on the method call. Therefore I've dropped the methods  `DoModal` and `Show` and introduced a new method `CGUIDialog::Open` as a replacement.

@Montellese @mkortstiege @Paxxi mind taking a look.

### CODE REVIEW

**Comment by mkortstiege** [comment]
Look good and clean to me. I wonder if there's a scenario where a dialog has to be called with a specific modality though.

**Comment by xhaggi** [comment]
any other objections? @Paxxi or @Montellese

**Comment by xhaggi** [comment]
jenkins build this please

**Comment by Paxxi** [comment]
@xhaggi I haven't gone through it in detail but I like the changes and cleanup and I think it's going in the right direction, :+1:

**Comment by xhaggi** [comment]
ok i'm waiting for some feedback from @Montellese and if he's fine we should shove it in asap.

File: xbmc/addons/AddonStatusHandler.cpp [hunk 1]
```diff
@@ -102,10 +102,7 @@ void CAddonStatusHandler::Process()
       pDialog->SetHeading(CVariant{heading});
       pDialog->SetLine(1, CVariant{24070});
       pDialog->SetLine(2, CVariant{24073});
-
-      //send message and wait for user input
-      ThreadMessage tMsg = {TMSG_DIALOG_DOMODAL, WINDOW_DIALOG_YES_NO, g_windowManager.GetActiveWindow()};
-      CApplicationMessenger::Get().SendMessage(tMsg, true);
+      pDialog->Open();
```
Comment (Montellese) [inline xbmc/addons/AddonStatusHandler.cpp]: The changes in this file look like a change in behaviour or not? Both are blocking calls but the old code was opening the dialog in the main thread and the new code opens it in the current thread.

File: xbmc/dialogs/GUIDialogBusy.cpp [hunk 2]
```diff
@@ -84,10 +86,10 @@ bool CGUIDialogBusy::WaitOnEvent(CEvent &event, unsigned int displaytime /* = 10
 }
 
 CGUIDialogBusy::CGUIDialogBusy(void)
-  : CGUIDialog(WINDOW_DIALOG_BUSY, "DialogBusy.xml"), m_bLastVisible(false)
+  : CGUIDialog(WINDOW_DIALOG_BUSY, "DialogBusy.xml", DialogModalityType::SYSTEM_MODAL),
+  m_bLastVisible(false)
```
Comment (Montellese) [inline xbmc/dialogs/GUIDialogBusy.cpp]: Indentation

**Comment by Montellese** [comment]
As mentioned in the commit comment I'm not 100% sure about all the changes from (blocking) threaded messages to directly opening the dialog since it changes the thread in which the dialog is opened. But I also don't know the guilib code that well.

File: xbmc/addons/AddonStatusHandler.cpp (hunk 1 above)
Comment (xhaggi) [inline xbmc/addons/AddonStatusHandler.cpp]: we check for the main thread in Open() and if not we send a thread msg which should be the same like this, or isn't it?

Comment (Montellese) [inline xbmc/addons/AddonStatusHandler.cpp]: You're right. I didn't know that Open had that kind of logic.

**Comment by xhaggi** [comment]
apart from you comments any other objections?

**Comment by Montellese** [comment]
I don't think so.

**Comment by xhaggi** [comment]
jenkins build and merge

**Comment by xhaggi** [comment]
jenkins build and merge

**Comment by xhaggi** [comment]
jenkins build this please

**Pull request merged**